runtime-manager/src/managers/mod.rs

```diff
@@ -154,108 +131,76 @@ impl ProtocolState {
     }
 
-    /// Invokes the entry point of the provisioned WASM program.  Will fail if
-    /// the current lifecycle state is not `LifecycleState::ReadyToExecute` or
-    /// if the WASM program fails at runtime.  On success, bumps the lifecycle
-    /// state to `LifecycleState::FinishedExecuting` and returns the error code
-    /// returned by the WASM program entry point as an `i32` value.
-    pub(crate) fn invoke_entry_point(&self) -> Result<i32, RuntimeManagerError> {
-        Ok(self.host_state.lock()?.invoke_entry_point()?)
+    //TODO: add description
+    pub(crate) fn append_file(&mut self, client_id: &Principal, file_name: &str, data: &[u8]) -> Result<(), RuntimeManagerError> {
+        self.is_modified = true;
+        self.vfs.lock()?.check_capability(client_id,file_name, &FileOperation::Write)?;
+        Ok(self.vfs.lock()?.append(file_name,data)?)
     }
 
-    /// Returns the current lifecycle state that the host provisioning state is
-    /// in.
-    pub(crate) fn get_lifecycle_state(&self) -> Result<LifecycleState, RuntimeManagerError> {
-        Ok(self.host_state.lock()?.get_lifecycle_state().clone())
+    //TODO: add description
+    pub(crate) fn read_file(&self, client_id: &Principal, file_name: &str) -> Result<Option<Vec<u8>>, RuntimeManagerError> {
+        self.vfs.lock()?.check_capability(client_id,file_name, &FileOperation::Read)?;
+        Ok(self.vfs.lock()?.read(file_name)?)
     }
 
-    /// Returns a result of a WASM computation that has executed on the host
-    /// provisioning state.  Returns `None` iff no such result has been
-    /// registered.
-    pub(crate) fn get_result(&self) -> Result<Option<Vec<u8>>, RuntimeManagerError> {
-        Ok(self.host_state.lock()?.get_result().map(|o| o.clone()))
+    /// Requests shutdown on behalf of a client, as identified by their client
+    /// ID, and then checks if this request was sufficient to reach a threshold
+    /// of requests wherein the platform can finally shutdown.
+    pub(crate) fn request_and_check_shutdown(
+        &mut self,
+        client_id: u64,
+    ) -> Result<bool, RuntimeManagerError> {
+        self.expected_shutdown_sources.retain(|v| v != &client_id);
+        Ok(self.expected_shutdown_sources.is_empty())
     }
+    
+    pub(crate) fn execute(&mut self, file_name: &str, client_id: u64) -> ProvisioningResult {
+        let execution_strategy = match self.global_policy.execution_strategy() {
+            ExecutionStrategy::Interpretation => {
+                execution_engine::factory::ExecutionStrategy::Interpretation
+            }
+            ExecutionStrategy::JIT => execution_engine::factory::ExecutionStrategy::JIT,
+        };
+        let return_code = execute(
+            &execution_strategy,
+            self.vfs.clone(),
+            file_name,
+        )?;
+        
+        let response = if return_code == EngineReturnCode::Success {
+            let result = self.read_file(&Principal::Participant(client_id),"output")?;
```

Comment:
Are these file name, e.g. "output"`, just hardcoded into the source throughout the code base?  Can we have them hoisted out as constants?